genet/core.py

```diff
@@ -2263,3 +2263,58 @@ def get_link_slope_dictionary(self, elevation_dict):
             slope_dict[link_id] = {'slope': link_slope}
 
         return slope_dict
+
+    def summary(self):
+        """
+        Returns a report with summary statistics for the network and the schedule.
+        """
+        logging.info('Creating a summary report')
+        report = {}
+
+        network_stats = {'Number of network links': nx.number_of_nodes(self.graph),
+                         'Number of network nodes': nx.number_of_edges(self.graph)}
+        report['network_graph_info'] = network_stats
+        report['modes'] = {}
+        report['modes']['Modes on network links'] = self.modes()
+
+        highway_tags = self.link_attribute_data_under_key({'attributes': 'osm:way:highway'})
+        highway_tags = set(itertools.chain.from_iterable(highway_tags.apply(lambda x: persistence.setify(x))))
+        osm_highway_tags = {}
+        for tag in highway_tags:
+            tag_links = self.extract_links_on_edge_attributes(conditions={'attributes': {'osm:way:highway': tag}},
+                                                              mixed_dtypes=True)
+            osm_highway_tags[tag] = len(tag_links)
+        report['osm_highway_tags'] = {'Number of links by tag': osm_highway_tags}
+
+        links_by_mode = {}
+        for mode in self.modes():
+            mode_links = self.extract_links_on_edge_attributes(conditions={'modes': mode}, mixed_dtypes=True)
+            links_by_mode[mode] = len(mode_links)
+        report['modes']['Number of links by mode'] = links_by_mode
+
+        if self.schedule:
+            schedule_stats = {'Number of services': self.schedule.__len__(),
+                              'Number of routes': self.schedule.number_of_routes(),
+                              'Number of stops:': len(self.schedule.reference_nodes())}
+            report['schedule_info'] = schedule_stats
+
+            schedule_modes = self.schedule.modes()
+            report['modes']['Modes in schedule'] = schedule_modes
+
+            services_by_modes = {}
+            for mode in schedule_modes:
+                services_by_modes[mode] = len(self.schedule.services_on_modal_condition(mode))
+            report['modes']['Services by mode'] = services_by_modes
+
+            stops_by_modes = {}
+            for mode in schedule_modes:
+                stops_by_modes[mode] = len(self.schedule.stops_on_modal_condition(mode))
+            report['modes']['PT stops by mode'] = stops_by_modes
+
+            report['accessibility_tags'] = {}
+            report['accessibility_tags']['Stops with tag bikeAccessible'] = len(
+                self.schedule.stop_attribute_data({'attributes': 'bikeAccessible'}))
+            report['accessibility_tags']['Stops with tag carAccessible'] = len(
+                self.schedule.stop_attribute_data({'attributes': 'carAccessible'}))
```

Comment:
here I would use that pandas object a bit more, dataframe or series, I don't remember anymore, apart from just the length, report also the unique values for those tags, which is something that Fred had issues with in the matsim config: whether the values are 'true' or 'True' etc.